chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -50,22 +57,28 @@ pub enum BlockTransactableRef<'a> {
 
 /// The change that a block has caused to the blockchain state
 pub struct TransactionVerifierDelta {
-    data: BTreeMap<OutPointSourceId, CachedInputsOperation>,
+    tx_index: BTreeMap<OutPointSourceId, CachedInputsOperation>,
+    utxo_cache: ConsumedUtxoCache,
+    utxo_block_undo: BTreeMap<Id<Block>, BlockUndoEntry>,
 }
 
 /// The tool used to verify transaction and cache their updated states in memory
 pub struct TransactionVerifier<'a, S> {
     db_tx: &'a S,
     tx_index_cache: BTreeMap<OutPointSourceId, CachedInputsOperation>,
+    utxo_cache: UtxosCache<'a>,
+    utxo_block_undo: BTreeMap<Id<Block>, BlockUndoEntry>,
     chain_config: &'a ChainConfig,
 }
 
-impl<'a, S> TransactionVerifier<'a, S> {
-    pub fn new(db_tx: &'a S, chain_config: &'a ChainConfig) -> Self {
+impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
```

Comment:
I don't think this bound that you added here is necessary. There are no operations related to it in this impl block.